build-support/common_shell_methods_test.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Hmmm, I think the test is overkill - this isn't user-facing, and `log` is a trivial function.

Reply:
Yes, the `log` test was just a stepping-stone to the `determine_python` test. I mainly wanted to test `determine_python` because the logic isn't exactly trivial, and because this broke when I was introducing someone else to developing in the pants repo, which was embarrassing. I would like to keep the `determine_python` test, but I have removed `log`.

Reply:
Having this test file also makes it clear how to add more test cases should we add/change any further functionality in the build-support scripts. Given that we already saw a regression without a test, I don't see why it's not worth it to keep this test here to ensure we don't move too fast and break things later.

Reply:
Well tests aren't free - they are code that requires maintenance, and add to CI time. Obviously most of the time we accept that tradeoff, but this is a script only used by Pants developers, not by end users, so the tradeoffs are different.

Reply:
And unless I'm missing something, this is still just testing logging?